### PR TITLE
Adapt to IMDG using UUID instead of Address for Invocations

### PR DESCRIPTION
**Pull request by jbartok**
Add description here

Checklist
- [ ] Tags Set
- [ ] Milestone Set
- [ ] Any breaking changes are documented
- [ ] New public APIs have `@Nonnull/@Nullable` annotations
- [ ] New public APIs have `@since` tags in Javadoc
- [ ] For code samples, code sample main readme is updated

Links to issues fixed (if any):

Fixes #NNNN

List of breaking changes:

* ..

### CODE REVIEW

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JetClientInstanceImpl.java [hunk 1]
```diff
@@ -151,9 +151,9 @@ public ILogger getLogger() {
         }
     }
 
-    private static Address masterAddress(Cluster cluster) {
+    private static UUID masterUUID(Cluster cluster) {
```
Comment (cangencer) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JetClientInstanceImpl.java]: can you make this a Util method? it's repeated several times.

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JobProxy.java [hunk 2]
```diff
@@ -142,6 +145,11 @@ protected JobConfig doGetJobConfig() {
     }
 
     @Override
+    protected UUID masterUuid() {
+        Optional<Member> first = container().getClusterService().getMembers().stream().findFirst();
```
Comment (sancar) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JobProxy.java]: We have a getMasterMember on the ClusterService method.

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/ClientJobProxy.java [hunk 3]
```diff
@@ -156,16 +155,15 @@ protected long doGetJobSubmissionTime() {
...
             return serializationService().toObject(data);
         });
     }
 
     @Override
-    protected Address masterAddress() {
-        Optional<Member> first = container().getCluster().getMembers().stream().findFirst();
-        return first.orElseThrow(() -> new IllegalStateException("No members found in cluster")).getAddress();
+    protected UUID masterUuid() {
+        return Util.getMaster(container().getCluster()).getUuid();
```
Comment (cangencer) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/ClientJobProxy.java]: you can use use clientClusterService.getMaster() here and inline the Util method

Comment (jbartok) [reply hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/ClientJobProxy.java]: Ok, but we should be aware that we are changing behaviour, not just adapting (ie. no more exception). Is that ok?

File: hazelcast-jet-all/src/main/java/com/hazelcast/jet/server/JetCommandLine.java [hunk 4]
```diff
@@ -406,7 +406,7 @@ public void cluster(
             JetClientInstanceImpl client = (JetClientInstanceImpl) jet;
             HazelcastClientInstanceImpl hazelcastClient = client.getHazelcastClient();
             ClientClusterService clientClusterService = hazelcastClient.getClientClusterService();
-            Member masterMember = clientClusterService.getMember(clientClusterService.getMasterAddress());
+            Member masterMember = clientClusterService.getMember(clientClusterService.getMasterMember().getUuid());
```
Comment (sancar) [inline hazelcast-jet-all/src/main/java/com/hazelcast/jet/server/JetCommandLine.java]: To get master member, we get the master member and get its uuid and then get the master member again ?
@jbartok

**Pull request merged**